### ISSUE TITLE
haskellPackages.clash-prelude is broken

### ISSUE THREAD

**Issue by rowanG077** [NixOS/nixpkgs#140613]
### Describe the bug
haskellPackages.clash-prelude is broken

### Steps To Reproduce
1. Run `nix-shell -p haskellPackages.clash-prelude`
2. Package is marked as broken. Allowing broken packages allows it to start compiling but it fails on `src/Clash/Annotations/TH.hs` source file with the following error:

```
[ 95 of 106] Compiling Clash.Annotations.TH ( src/Clash/Annotations/TH.hs, dist/build/Clash/Annotations/TH.p_o )

<no location info>: error:

ByteCodeLink.lookupCE
During interactive linking, GHCi couldn't find the following symbol:
  recursionzmschemeszm5zi2zi2zi1zmYW4xOyDGcg3POedRDdiKO_DataziFunctorziFoldableziTH_zdfMakeBaseFunctorNamezuzdcmakeBaseFunctor_closure
This may be due to you not asking GHCi to load extra object files,
archives or DLLs needed by your current session.  Restart GHCi, specifying
the missing library using the -L/path/to/object/dir and -lmissinglibname
flags, or simply by naming the relevant files on the GHCi command line.
Alternatively, this link failure might indicate a bug in GHCi.
If you suspect the latter, please report this as a GHC bug:
  https://www.haskell.org/ghc/reportabug
```

This is a nix specific issue since I can build this package fine using stack. 

### Notify maintainers
@maralorn I doubt you are the right person but you touched it last.

### Metadata
``
`> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.14.7, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210922_bcd73eb`
 - channels(rowan.goemans): `"nixgl"`
 - channels(root): `"nixos-21.11pre319271.31ffc50c571"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by maralorn** [comment]
Yeah, I am not sure how I touched it. But I don‘t assume any responsibility for it. This is an error while executing TH.
The fix is probably simple, but I don‘t know it.
